File: QuickStarts/VB/EventBroker/Tests/Tests.SimpleEventBroker/TestObjects/ClipboardMonitor.vb
'===============================================================================
' Microsoft patterns & practices
' Unity Application Block
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================

Imports System

Namespace EventBrokerTests.TestObjects
	Class ClipboardMonitor
    Private _numberOfCopyOperations As Integer
    Private _numberOfPasteOperations As Integer

		Public Sub New()
      _numberOfPasteOperations = 0
      _numberOfCopyOperations = 0
		End Sub

		Public ReadOnly Property NumberOfCopyOperations() As Integer
			Get
        Return _numberOfCopyOperations
			End Get
		End Property

		Public ReadOnly Property NumberOfPasteOperations() As Integer
			Get
        Return _numberOfPasteOperations
			End Get
		End Property

		Public Sub OnClipboardCopy(ByVal sender As Object, ByVal e As EventArgs)
      System.Threading.Interlocked.Increment(_numberOfCopyOperations)
		End Sub

		Public Sub OnClipboardPaste(ByVal sender As Object, ByVal e As EventArgs)
      System.Threading.Interlocked.Increment(_numberOfPasteOperations)
		End Sub
	End Class
End Namespace

File: QuickStarts/CS/EventBroker/Tests/Tests.SimpleEventBroker/TestObjects/ClipboardMonitor.cs
//===============================================================================
// Microsoft patterns & practices
// Unity Application Block
//===============================================================================
// Copyright © Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================

using System;

namespace EventBrokerTests.TestObjects
{
    class ClipboardMonitor
    {
        private int numberOfCopyOperations;
        private int numberOfPasteOperations;

        public ClipboardMonitor()
        {
            numberOfPasteOperations = 0;
            numberOfCopyOperations = 0;
        }

        public int NumberOfCopyOperations
        {
            get { return numberOfCopyOperations; }
        }

        public int NumberOfPasteOperations
        {
            get { return numberOfPasteOperations; }
        }

        public void OnClipboardCopy(object sender, EventArgs e)
        {
            ++numberOfCopyOperations;
        }

        public void OnClipboardPaste(object sender, EventArgs e)
        {
            ++numberOfPasteOperations;
        }
    }
}
